packages/http-specs/specs/response/status-code-range/main.tsp

```diff
@@ -0,0 +1,45 @@
+import "@typespec/http";
+import "@typespec/spector";
+
+using Http;
+using Spector;
+
+/**
+ * Test for range of status code.
+ */
+@scenarioService("/response/status-code-range")
+namespace Response.StatusCodeRange;
+
+@scenario
+@scenarioDoc("""
+  Test case for range of status code in error response.
+  
+  Verify that the result of the API is an error/exception in client, and the error response can be de-serialized to ErrorInRange model (instead of DefaultError model).
+  
+  Expected status code 494 and response body:
+  ```json
+  { 
+    "code": "request-header-too-large",
+    "message": "Request header too large"
+  }
+  ```
+  """)
+@route("/error-response")
+@get
+op errorResponse(): 204 | ErrorInRange | DefaultError;
```

Comment:
do we need to test overlap case? for example, 404 error | other 4xx error.

Reply:
Yeah, I can add one.

Basically, it be `NotFoundResponse | Standard4XXResponse`, and the 2 error model be different.
Then, mock api returns 404 with `NotFoundResponse` model, right?

Reply:
correct.

Reply:
Does that we will have precedence in which the concrete status code should be processed with range error like: concrete error > status code range > default error code range?

Reply:
I am not sure whether there is agreement on this. Though I kind of think it be reasonable in this specific case. However I cannot say whether I know what to do on e.g. `204 | Error400to460 | Error440to480`